modern/clang/Dockerfile

```diff
@@ -0,0 +1,154 @@
+ARG DOCKER_USERNAME
```

Comment:
so, this is not an actual `clang` image, but instead, it's just an intermediate builder image, which we don't expect to be used by consumers, correct?
maybe then rename it to the `clang-builder` to avoid confusion? it might not be so clear which Dockerfile do I need to modify, especially for newcomers

Reply:
Correct, it's an intermediate layer which builds the compiler. Okay, clang-builder sounds a good name.